pulsar-common/src/main/java/org/apache/pulsar/common/util/anonymizer/DefaultAuthenticationRoleLoggingAnonymizer.java

```diff
@@ -24,36 +24,41 @@
  * This class provides a utility to anonymize authentication roles before logging,
  * based on a configured anonymization strategy. The anonymization strategy is
  * determined by the provided value of the {@link DefaultRoleAnonymizerType} enum.
- *
+ * <p>
  * The primary purpose of this class is to enable flexible anonymization of sensitive
  * information (such as user roles) in logs, ensuring compliance with security and
  * privacy requirements while allowing customization of the anonymization behavior.
- *
+ * <p>
  * Usage:
  * - The class constructor accepts a string that represents the desired anonymization
  *   strategy (e.g., "NONE", "REDACTED", "SHA256", "MD5"), and it initializes the
  *   anonymizer type accordingly.
  * - The {@code anonymize} method applies the selected anonymization strategy to
  *   the provided role and returns the anonymized value.
- *
+ * <p>
  * Example:
  * <pre>
  * DefaultAuthenticationRoleLoggingAnonymizer roleAnonymizer =
  *     new DefaultAuthenticationRoleLoggingAnonymizer("SHA256");
  * String anonymizedRole = roleAnonymizer.anonymize("admin");
  * </pre>
- *
+ * <p>
  * Anonymization strategies:
  * - NONE: No anonymization (returns the role as-is).
  * - REDACTED: Replaces the role with "[REDACTED]".
- * - SHA256: Hashes the role using the SHA-256 algorithm.
- * - MD5: Hashes the role using the MD5 algorithm.
+ * - hash:SHA256: Hashes the role using the SHA-256 algorithm and prefixes it with "SHA-256:".
+ * - hash:MD5: Hashes the role using the MD5 algorithm and prefixes it with "MD5:"
  */
 public final class DefaultAuthenticationRoleLoggingAnonymizer {
    private static DefaultRoleAnonymizerType anonymizerType = NONE;
 
    public DefaultAuthenticationRoleLoggingAnonymizer(String authenticationRoleLoggingAnonymizer) {
-      anonymizerType = DefaultRoleAnonymizerType.valueOf(authenticationRoleLoggingAnonymizer);
+      if (authenticationRoleLoggingAnonymizer.startsWith("hash:")) {
+         anonymizerType = DefaultRoleAnonymizerType.valueOf(authenticationRoleLoggingAnonymizer
+                 .substring("hash:".length()));
+      } else {
+         anonymizerType = DefaultRoleAnonymizerType.valueOf(authenticationRoleLoggingAnonymizer);
```

Comment:
do `.toUpperCase()` for the String parameter passed to `.valueOf` so that the parameter is case in-sensitive.